data/src/main/kotlin/com/vultisig/wallet/data/api/EvmApi.kt

```
@@ -180,6 +203,102 @@ class EvmApiImp(private val http: HttpClient, private val rpcUrl: String) : EvmA
+    private suspend fun fetchBalancesPerToken(
+        address: String,
+        contractAddresses: List<String>,
+    ): Map<String, BigInteger> = coroutineScope {
+        contractAddresses
+            .map { contract ->
+                async {
+                    contract to
+                        try {
+                            if (contract.isEmpty()) getETHBalance(address)
+                            else getERC20Balance(address, contract)
+                        } catch (e: SocketTimeoutException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)